### PR TITLE
Add management of iaas via docker machine

### PR DESCRIPTION
**Pull request by gtsalles**
This PR is intended to replace the management of digital ocean iaas for the usage of libmachine (a part of [docker machine](https://github.com/docker/machine/)) allowing to remove a bunch of hard coded stuff (iteraction with digital ocean API, management of ssh keys), and making possible the addition of new providers easily.

This also adds makes all connections to docker to be established via TLS, since this is a default behavior of docker machine.

Related to #54

### CODE REVIEW

File: iaas/digitalocean/digitalocean.go [hunk 1]
```diff
@@ -1,465 +1,118 @@
...
-		id, _ := strconv.Atoi(snapshot.ID)
-		image = godo.DropletCreateImage{
-			ID: id,
-		}
-	}
-	sshKey, err := do.getSSHKeyForDroplet()
-	if err != nil {
-		return
-	}
+func New(token string) (do *provider, err error) {
```
Comment (golangci) [inline iaas/digitalocean/digitalocean.go]: exported func New returns unexported type *digitalocean.provider, which can be annoying to use

File: iaas/iaas.go [hunk 2]
```diff
@@ -42,4 +14,5 @@ type Machine struct {
 	Name      string `json:"name"`
 	Kind      string `json:"kind"`
 	SSHKeysID []int  `json:"ssh_keys_id"`
+	CertsDir  string `json:certs_dir`
```
Comment (golangci) [inline iaas/iaas.go]: struct field tag `json:certs_dir` not compatible with reflect.StructTag.Get: bad syntax for struct tag value

File: iaas/digitalocean/digitalocean.go [hunk 3]
```diff
@@ -1,465 +1,118 @@
...
+	Client            libmachine.API
+	Host              *host.Host
+	Name              string
+	ClientPath        string
+	Region            string
+	Size              string
+	ImageSlug         string
+	KeyID             int
+	Ctx               context.Context
+	sshPublicKeyPath  string
```
Comment (golangci) [inline iaas/digitalocean/digitalocean.go]: `sshPublicKeyPath` is unused

File: iaas/digitalocean/digitalocean.go [hunk 4]
```diff
@@ -1,465 +1,118 @@
...
+	sshPrivateKeyPath string
```
Comment (golangci) [inline iaas/digitalocean/digitalocean.go]: `sshPrivateKeyPath` is unused

File: examples/concurrent/main.go [hunk 5]
```diff
@@ -49,9 +50,16 @@ func run(contextDir, dockerfile, imageName, remoteBuildURI, volumeSource, volume
 		containerOpts.Volumes = []string{fmt.Sprintf("%s:%s", volumeSource, volumeDestination)}
 	}
 	if remote {
-		buildOpts.Iaas = &digitalocean.Digitalocean{
-			Ctx: context.Background(),
+		key := os.Getenv("DIGITALOCEAN_API_KEY")
+		if key == "" {
+			log.Println("You must provide an api key for digital ocean")
```
Comment (marioidival) [inline examples/concurrent/main.go]: If you use `log.Fatal`, the process stop here, make sense?

**Comment by avelino** [comment]
Fixed permission

**Pull request merged**